jupyter_client/multikernelmanager.py

```diff
@@ -90,7 +90,7 @@ def start_kernel(self, kernel_name=None, **kwargs):
 
         The kernel ID for the newly started kernel is returned.
         """
-        kernel_id = kwargs.pop('kernel_id', unicode_type(uuid.uuid4()))
+        kernel_id = self.determine_kernel_id(**kwargs)
```

Comment:
Should `determine_kernel_id` be called if kernel_id is specified? It seems like it should only be called if kernel_id is not specified.

Reply:
@minrk - thank you for taking a look at this.

I was originally thinking that the override should be called unconditionally with the base implementation honoring the existence of the parameter, but that probably leaves too much room for mistakes, so I'd be okay with calling the method only when `kernel_id is None`.

Not sure if its my Java background (very new to python), but I don't like using `new` as a verb.  Does something like `init_kernel_id(**kwargs)` sound any better?   

Would you mind suggesting the appropriate *python way* for constructing the conditional call based on the `pop` result?

Reply:
I would write it like this:

```python
if 'kernel_id' in kwargs:
    kernel_id = kwargs.pop('kernel_id')
else:
    kernel_id = self.new_kernel_id()
```

By Python (or maybe just Jupyter) convention, I like `new` better than `init`, because `init` implies a stateful change. I would expect `.init_kernel_id()` to *set* `self.kernel_id`, not return a value I'm free to use. The verb I would use for a stateless callable that returns a new kernel id is 'new'. 'generate' works, too, but it's more verbose than I would like.

Reply:
Thanks for the explanation on `new` vs. `init` - that makes sense.  Regarding the example (and a better understanding of python args), I figured things would be more compact and  something like the original would have been suggested:
```python
kernel_id = kwargs.pop('kernel_id', self.new_kernel_id(**kwargs))
```
I'd prefer to push this version, but would be happy to submit the expanded form if you feel strongly about it.

Reply:
It doesn't make much of a difference here, but I find clarity is typically preferable to compactness. The `if/else` is also ever-so-slightly more efficient, since `new_kernel_id` is only called if needed, rather than this PR, where it's called unconditionally, even if the result is just discarded.